src/docs/development/platform-integration/add-to-app-android/add-flutter-screen.md

```diff
@@ -0,0 +1,266 @@
+Flutter manually. This initialization must occur before you attempt to display any Flutter
+content, or execute any Dart code.
+
+To manually initialize Flutter, execute the following code at the desired location:
+
+```java
+FlutterMain.startInitialization(appContext);
+FlutterMain.ensureInitializationComplete(appContext, null);
+```
+
+The call to `ensureInitializationComplete()` is a blocking call. Make sure you invoke it at a
+place and time that can afford a brief processing pause.
+
+### Step 3: Add FlutterActivity to AndroidManifest.xml
+
+Flutter provides `FlutterActivity` to display a Flutter experience within an Android `Activity`.
+Like any other `Activity`, `FlutterActivity` must be registered in your `AndroidManifest.xml`.
+Add the following XML to your `AndroidManifestxml` file under your `application` tag:
+
+```xml
+<activity
+  android:name=".YourFlutterActivityName"
+  android:launchMode="singleTop"
+  android:theme="@style/LaunchTheme"
+  android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
+  android:hardwareAccelerated="true"
+  android:windowSoftInputMode="adjustResize"
+  />
+```
+
+Replace `YourFlutterActivityName` with the name of your Flutter `Activity`.
+
+### Step 4: Launch FlutterActivity
+
+With `FlutterActivity` registered in your manifest file, add code to launch `FlutterActivity`
+from whatever point in your app that you'd like. The following example shows `FlutterActivity`
+being launched from an `OnClickListener`.
+
+```java
+myButton.addOnClickListener(new OnClickListener() {
+  @Override
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity.createBuilder().build(currentActivity)
+    );
+  }
+});
+```
+
+The above example assumes that your Dart entrypoint is called `main()`, and your initial
+Flutter route is '/'. If you'd like to change either property, use the `IntentBuilder` to 
+provide your desired properties:
+
+```java
+myButton.addOnClickListener(new OnClickListener() {
+  @Override
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity
+        .createBuilder()
+        .dartEntrypoint("mySpecialEntrypoint")
+        .initialRoute("/my_route")
+        .build(currentActivity)
+      );
+  }
+});
+```
+
+Replace `"mySpecialEntrypoint"` with the name of your desired Dart entrypoint, and replace
+`"/my_route"` with your desired initial route.
+
+### Step 5: (Optional) Use a cached FlutterEngine
+
+Every `FlutterActivity` creates its own `FlutterEngine` by default. Each `FlutterEngine` has
+a non-trivial "warm-up" time. This means that launching a standard `FlutterActivity` comes 
+with a brief delay before your Flutter experience becomes visible. To avoid this delay, you
+can warm-up a `FlutterEngine` before arriving at your `FlutterActivity` and then you can use
+your pre-warmed `FlutterEngine` instead.
+
```

Comment:
We didn't write the page that comes before this page yet but since we're going to push the staging site at the same time, we can pre-emptively write this:

Add an info box that says the debug/release builds of Flutter has drastically different performance characteristics. To evaluate the performance of Flutter, use a Flutter release build.